test/com/dataengi/crm/identities/daos/UsersDAOSpec.scala

```diff
@@ -24,42 +24,195 @@ class UsersDAOSpec
 
     "add user" in {
-      val NewCompanyName: String          = "NEW_COMPANY" + Gen.alphaStr.sample.get
-      val newCompanyResult: XorType[Long] = companiesService.create(NewCompanyName).await()
-      val newCompany                      = companiesService.get(newCompanyResult.value).await().value
-      val role                            = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
 
-      val user = User(
+      val role                     = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
+      val company                  = createCompany()
+      val TestLoginInfo: LoginInfo = LoginInfo("provider", "dasd@asd.com")
+      val user: User = User(
         loginInfo = TestLoginInfo,
-        company = newCompany,
+        company = company,
         role = role
       )
 
       val getUserResult = usersDAO.add(user).await()
       println(s"[user-dao][get] ${getUserResult.logResult}")
       getUserResult.isRight === true
+
+    }
+
+    "add userList" in {
+      val company                   = createCompany()
+      val role                      = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
+      val TestLoginInfo: LoginInfo  = LoginInfo("provider", "dasd@asd.com")
+      val TestLoginInfo2: LoginInfo = LoginInfo("provider2", "dasd2@asd.com")
+      val user = User(
+        loginInfo = TestLoginInfo,
+        company = company,
+        role = role
+      )
+      val user2 = User(
+        loginInfo = TestLoginInfo2,
+        company = company,
+        role = role
+      )
+
+      val userList = List(user, user2)
+
+      company.id must not be None
+
+      val companyId = company.id.get
+
+      val getUserResult = usersDAO.add(userList).await()
+      println(s"[user-dao][get] ${getUserResult.logResult}")
+      getUserResult.isRight === true
+
+      val users = usersDAO.findByCompany(companyId).await().value
+
+      users must have size 2
+
+      users.filter((user: User) => user.loginInfo.providerID == "provider") must have size 1
+      users.filter((user: User) => user.loginInfo.providerID == "provider2") must have size 1
+
     }
 
     "find user by login info" in {
+
+      val TestLoginInfo: LoginInfo = LoginInfo("provider", "dasd@asd.com")
+
       val findUserResult = usersDAO.find(TestLoginInfo).await()
       println(s"[user-dao][find] ${findUserResult.logResult}")
       findUserResult.isRight === true
       val user = findUserResult.value.get
       user.loginInfo === TestLoginInfo
     }
 
+    "find user by companyId negative" in {
+      val company = createCompany()
+      val role    = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
+
+      company.id must not be None
+
+      val companyId      = company.id.get
+      val findUserResult = usersDAO.findByCompany(companyId).await()
+      println(s"[user-dao][find] ${findUserResult.logResult}")
+      findUserResult.isRight === true
```

Comment:
if you have to check of some result value, you don't need always check is it `right`